Sources/TuistGenerator/Linter/TargetLinter.swift

```diff
@@ -292,6 +293,51 @@ class TargetLinter: TargetLinting {
         }
         return []
     }
+
+    private func lintAllRequiredFilesAvailable(target: Target) -> [LintingIssue] {
+        var lintingIssues: [LintingIssue] = []
+
+        for path in target.sourcesPaths {
+            if target.sources.isEmpty, !path.isOpaqueDirectory {
+                lintingIssues.append(LintingIssue(reason: "No files found at: \(path)", severity: .warning))
+                continue
+            }
+            guard target.sources.first(where: { $0.path == path }) != nil else {
+                lintingIssues.append(LintingIssue(reason: "No files found at: \(path)", severity: .warning))
+                continue
+            }
+        }
+
+        if target.supportsResources {
+            for path in target.resourcesPaths {
+                if target.resources.resources.isEmpty {
+                    lintingIssues.append(LintingIssue(reason: "No resources found at: \(path)", severity: .warning))
+                    continue
+                }
+                guard target.resources.resources.first(where: { $0.path == path }) != nil else {
+                    lintingIssues.append(LintingIssue(reason: "No resources found at: \(path)", severity: .warning))
+                    continue
+                }
+            }
+        }
+
+        if target.supportsResources {
+            let files = target.copyFiles.flatMap(\.files)
+
```

Comment:
We can lint based on glob, and folder reference too, but for now I made it simple for the sake of reviewing it